packages/path_provider/path_provider/README.md

```diff
@@ -19,3 +20,18 @@ String appDocPath = appDocDir.path;
 ```
 
 Please see the example app of this plugin for a full example.
+
+### Usage in tests
+
+Recently `path_provider` was updated to be a federated plugin. 
```

Comment:
Federated plugins and platform interface were bundled in how they were rolled out, but aren't actually intrinsically tied together. A monolithic plugin could still use a platform interface to allow some platforms to do more in Dart. (Also, minor wording nit; "recently" isn't going to continue to be true, but it's unlikely to be changed, so should be avoided).

How about: "`path_provider` now uses a `PlatformInterface `, meaning that not all platforms share the a single `PlatformChannel`-based implementation. With that change, tests should be updated to mock `PathProviderPlatform` rather than `PlatformChannel`."